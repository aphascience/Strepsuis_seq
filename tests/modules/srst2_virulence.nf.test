nextflow_process {

    name "Test Process SRST2_VIRULENCE"
    script "modules/srst2_virulence.nf"
    process "SRST2_VIRULENCE"

    test("Should map reads to virulence reference") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = ["test", file("${projectDir}/tests/data/input_S999_R1_001.fastq.gz"), file("${projectDir}/tests/data/input_S999_R2_001.fastq.gz")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
