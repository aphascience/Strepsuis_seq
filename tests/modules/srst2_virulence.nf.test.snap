{
    "Should map reads to virulence reference": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "virulence_test__genes__Ssuis_virulence__results.txt:md5,8622c9ba45892102cd2167ea78176c35"
                    ]
                ],
                "1": [
                    [
                        "test",
                        "virulence_test__fullgenes__Ssuis_virulence__results.txt:md5,84cce8e3f3842ba464bcbd603664276c"
                    ]
                ],
                "virulence_fullgenes": [
                    [
                        "test",
                        "virulence_test__fullgenes__Ssuis_virulence__results.txt:md5,84cce8e3f3842ba464bcbd603664276c"
                    ]
                ],
                "virulence_genes": [
                    [
                        "test",
                        "virulence_test__genes__Ssuis_virulence__results.txt:md5,8622c9ba45892102cd2167ea78176c35"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-24T10:25:50.934829178"
    }
}